pkg/controller/job/indexed_job_utils_test.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package job
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	batch "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestCalculateCompletedIndexesStr(t *testing.T) {
+	cases := map[string][]indexPhase{
+		"5,10": {
+			{2, v1.PodFailed},
+			{5, v1.PodSucceeded},
+			{5, v1.PodSucceeded},
+			{10, v1.PodFailed},
+			{10, v1.PodSucceeded},
+		},
+		"2,3,5-7": {
+			{0, v1.PodRunning},
+			{1, v1.PodPending},
+			{2, v1.PodSucceeded},
+			{3, v1.PodSucceeded},
+			{5, v1.PodSucceeded},
+			{6, v1.PodSucceeded},
+			{7, v1.PodSucceeded},
+		},
+		"0-2": {
+			{0, v1.PodSucceeded},
+			{1, v1.PodSucceeded},
+			{2, v1.PodSucceeded},
+			{3, v1.PodFailed},
+		},
+		"0,2-5": {
+			{0, v1.PodSucceeded},
+			{1, v1.PodFailed},
+			{2, v1.PodSucceeded},
+			{3, v1.PodSucceeded},
+			{4, v1.PodSucceeded},
+			{5, v1.PodSucceeded},
+		},
```

Comment:
add unit test with just one pod (that succeeds)

Reply:
👍 added